CONTRIBUTING.md

```diff
@@ -19,6 +19,29 @@ reported the issue. Please try to include as much information as you can. Detail
 * Any modifications you've made relevant to the bug
 * Anything unusual about your environment or deployment
 
+## Development Environment
+
+1. You will need Python 2 >= 2.7 or Python 3 >= 3.4. Verify which version you have by running `which python`. 
+
+    > If you don't have it installed, download it [here](https://www.python.org/downloads/). When you do this, `pip` should be installed automatically.
+
+1. Install `cfn-lint` from source by cloning the repository from Github and doing a developer install:
+
+    ```bash
+    git clone https://github.com/aws-cloudformation/cfn-python-lint.git
+    cd cfn-python-lint
+    pip install -e . # or "pip3 install -e ." if you are using Python 3.4+
+    ```
+
+1. Run `pip show cfn-lint` (or `pip3 show cfn-lint` if you are using Python 3). The `Location` printed should be the folder from the step above.
+1. You should now be able to modify the source code and see the changes immediately by running any of the `cfn-lint` commands.
+1. To ensure the tests pass, run:
+
+    ```bash
+    pip install mock # or "pip3 install mock"
+    python -m unittest discover ./test/
+    ```
```

Comment:
I usually push somewhere where Travis runs the tests for me because the [test matrix includes more tests in more Python versions as well](https://github.com/aws-cloudformation/cfn-python-lint/blob/master/.travis.yml), but documentation on running unit tests locally is still useful for debugging

Reply:
And I usually just run ‘tox -e py37’ or substitute for another test as needed.

Reply:
Oh i didn't know about tox. Will include!